go-controller/pkg/node/node.go

```diff
@@ -201,40 +208,43 @@ func (n *OvnNode) Start(wg *sync.WaitGroup) error {
 		return fmt.Errorf("timed out waiting for node's: %q logical switch: %v", n.name, err)
 	}
 
-	klog.Infof("Node %s ready for ovn initialization with subnet %s", n.name, util.JoinIPNets(subnets, ","))
+	if config.OvnKubeNode.Mode != types.NodeModeSmartNICHost {
+		klog.Infof("Node %s ready for ovn initialization with subnet %s", n.name, util.JoinIPNets(subnets, ","))
 
-	if _, err = isOVNControllerReady(n.name); err != nil {
-		return err
-	}
+		if _, err = isOVNControllerReady(n.name); err != nil {
+			return err
+		}
 
-	nodeAnnotator := kube.NewNodeAnnotator(n.Kube, node)
-	waiter := newStartupWaiter()
+		nodeAnnotator := kube.NewNodeAnnotator(n.Kube, node)
+		waiter := newStartupWaiter()
 
-	// Initialize management port resources on the node
-	mgmtPortConfig, err := createManagementPort(n.name, subnets, nodeAnnotator, waiter)
-	if err != nil {
-		return err
-	}
+		// Initialize management port resources on the node
+		mgmtPortConfig, err = createManagementPort(n.name, subnets, nodeAnnotator, waiter)
+		if err != nil {
+			return err
+		}
 
-	// Initialize gateway resources on the node
-	if err := n.initGateway(subnets, nodeAnnotator, waiter, mgmtPortConfig); err != nil {
-		return err
-	}
+		// Initialize gateway resources on the node
+		if err := n.initGateway(subnets, nodeAnnotator, waiter, mgmtPortConfig); err != nil {
+			return err
+		}
 
-	if err := nodeAnnotator.Run(); err != nil {
-		return fmt.Errorf("failed to set node %s annotations: %v", n.name, err)
-	}
+		if err := nodeAnnotator.Run(); err != nil {
+			return fmt.Errorf("failed to set node %s annotations: %v", n.name, err)
+		}
+		go n.gateway.Run(n.stopChan, wg)
```

Comment:
you move go n.gateway.Run to here on propose? in the old code it first wait for management port

Reply:
a mistake, it should be after `waiter` was invoked , good catch !